src/vg/civcraft/mc/citadel/listener/EntityListener.java

```diff
@@ -154,6 +179,271 @@ public void PlayerPromoteEvent(PromotePlayerEvent event){
+	                StringBuilder sb;
+	                if (player.hasPermission("citadel.admin.ctinfodetails")) {
+	                    sendAndLog(player, ChatColor.GREEN, String.format(
+	                        "Loc[%s]", pr.getLocation().toString()));
+	                    String groupName = "!NULL!";
+	                    if (group != null) {
+	                    	groupName = String.format("[%s] (%s)", group.getName(),
+	                    			group.getType().name());
+	                    }
+	                    sb = new StringBuilder();
+	                    sb.append(String.format(" Group%s Durability[%d/%d]",
+	                        groupName,
+	                        pr.getDurability(),
+	                        ReinforcementType.getReinforcementType
+	                        (pr.getStackRepresentation()).getHitPoints()));
+	                    int maturationTime = timeUntilMature(pr);
+	                    if (maturationTime != 0) {
+	                        sb.append(" Immature[");
+	                        sb.append(maturationTime);
+	                        sb.append("]");
+	                    }
+						int acidTime = timeUntilAcidMature(pr);
+						if (CitadelConfigManager.getAcidBlock() == block.getType()) {
+							sb.append(" Acid ");
+							if (acidTime != 0) {
+								sb.append("Immature[");
+								sb.append(acidTime);
+								sb.append("]");
+							} else {
+								sb.append("Mature");
+							}
+						}
+	                    if (pr.isInsecure()) {
+	                        sb.append(" (Insecure)");
+	                    }
+	                    if (group.isDisciplined()) {
+	                        sb.append(" (Disciplined)");
+	                    }
+	                    sb.append("\nGroup id: " + pr.getGroupId());
+	                    
+	                    sendAndLog(player, ChatColor.GREEN, sb.toString());
+	                    event.setCancelled(is_cancelled);
+	                    return;
+	                }
+                }
+        	}
+            boolean admin_bypass = player.hasPermission("citadel.admin.bypassmode");   
+        	if (state.isBypassMode() && (pr.isBypassable(player) || admin_bypass) && !pr.getGroup().isDisciplined()) {
+                 reinforcementBroken(player, rein);
+                 is_cancelled = false;
+            } else {
+            	
+            	ReinforcementDamageEvent dre = new ReinforcementDamageEvent(rein, player, event.getEntity().getLocation().getBlock());
+            	
+            	Bukkit.getPluginManager().callEvent(dre);
+            	
+            	if(dre.isCancelled()) {
+            		is_cancelled = true;
+            	}
+            	else {
+            		is_cancelled = reinforcementDamaged(player, rein);
+            	}
+            }
+            if (!is_cancelled) {
+                // The player reinforcement broke. Now check for natural
+                is_cancelled = createNaturalReinforcement(block, player) != null;
```

Comment:
Hmm, I guess it does have a purpose then, even though we aren't using any natural reinforcements.